packages/react-components/react-breadcrumb/docs/Spec.md

```diff
@@ -2,54 +2,479 @@
+
+#### DOM
+
+```HTML
+<nav aria-label="breadcrumb">
+  <ol>
+  {children}
+  </ol>
+</nav>
+```
+
+#### Breadcrumb structure:
+
+- root `nav` element
+- list `ol` element
+- `divider` - a BreadcrumbDivider component
+- `menu` - a BreadcrumbMenu component
+
+#### API
+
+| Property          | Values                                  | Default       | Purpose                                                                                  |
+| ----------------- | --------------------------------------- | ------------- | ---------------------------------------------------------------------------------------- |
+| appearance        | `transparent`, `subtle`                 | `transparent` | Sets appearance                                                                          |
+| menu              | _slot_                                  | undefined     | Sets custom dropdown menu                                                                |
+| focusMode         | `off`, `no-tab`, `tab-exit`, `tab-only` | `off`         | Sets the focus behavior for the Breadcrumb.                                              |
+| maxDisplayedItems | number                                  | undefined     | If items count more than `maxDisplayedItems` only this number of items will be displayed |
+| overflowIndex     | number                                  | 1             | Index of item where BreadcrumbMenu should be shown                                       |
+| size              | `small`, `medium`, `large`              | `medium`      | Defines size of the Breadcrumb                                                           |
+
+### BreadcrumbItem
+
+#### Anatomy
+
+![visual anatomy of the BreadcrumbItem component](./assets/breadcrumb-item-anatomy.png)
+
+BreadcrumbItem can be:
+
+- Button
+- Link
+- Non-clickable content
+- Dropdown Menu
+- Icon (optional)
+
+It can contain a tooltip.
+
+#### DOM
+
+Non-clickable element
+
+```HTML
+<li>
+  {children}
+</li>
+```
+
+Link
+
+```HTML
+<li>
+  <a href="#">
+    {children}
+  </a>
+</li>
+```
+
+Button
+
+```HTML
+<li>
+  <button>
+    {children}
+  </button>
+</li>
+```
+
+Usage
+
+```jsx
+<BreadcrumbItem>
+  Item 1
+</BreadcrumbItem>
+<BreadcrumbItem onClick={() => console.log('smth...')}>
+  Item 2
+</BreadcrumbItem>
+<BreadcrumbItem href="#">
+  Item 3
+</BreadcrumbItem>
+```
+
+#### API
+
+| Property     | Values                                  | Default   | Purpose                                     |
+| ------------ | --------------------------------------- | --------- | ------------------------------------------- |
+| active       | boolean                                 | false     | Indicates if the item is active             |
+| disabled     | boolean                                 | false     | Disables Breadcrumb item                    |
+| divider      | _slot_                                  | `chevron` | Sets type of divider                        |
```

Comment:
How are you planning to handle the divider here?
Are those two options valid?
```tsx
<BreadcrumbItem divider="slash">Hello</BreadcrumbItem>
```
```tsx
<BreadcrumbItem divider={<MyCustomDivider />}>Hello</BreadcrumbItem>
```

I'm not sure how it would be to handle both types in one prop. It might be worth to check how the TypeScript types would work if the same prop accepted a component or string. :)

Reply:
It'll be like a slot. Here's an implementation: https://github.com/microsoft/fluentui/pull/27199

Reply:
There are valid options:
```jsx
<BreadcrumbItem divider={<ArrowRight16Filled />}>Item with custom divider</BreadcrumbItem>
<BreadcrumbItem divider={{ variant: 'slash', size: 'small' }}>Item with slash divider</BreadcrumbItem>
<BreadcrumbItem divider={{ variant: null }}>Item with no divider</BreadcrumbItem>
```

Reply:
Divider will be used as a separate component, not as a slot because of tree shaking.